hail/python/hailtop/hailctl/dataproc/connect.py

```diff
@@ -38,44 +37,52 @@ def main(args, pass_through_args):  # pylint: disable=unused-argument
     }
     connect_port_and_path = dataproc_port_and_path[service]
 
-    # open SSH tunnel to master node
-    sp.check_call(
-        ['gcloud',
-         'compute',
-         'ssh',
-         '{}-m'.format(args.name),
-         '--zone={}'.format(args.zone),
-         '--ssh-flag=-D {}'.format(args.port),
-         '--ssh-flag=-N',
-         '--ssh-flag=-f',
-         '--ssh-flag=-n'],
-        stderr=sp.STDOUT
-    )
+    cmd = ['gcloud',
+           'compute',
+           'ssh',
+           '{}-m'.format(args.name),
+           '--zone={}'.format(args.zone),
+           '--ssh-flag=-D {}'.format(args.port),
+           '--ssh-flag=-N',
+           '--ssh-flag=-f',
+           '--ssh-flag=-n']
+
+    print('gcloud command:')
+    print(' '.join(cmd[:4]) + ' \\\n    ' + ' \\\n    '.join(cmd[4:]))
```

Comment:
```suggestion
    print(' '.join(cmd[:4]) + ' \\\n    ' + ' \\\n    '.join(f"'{x}'" for x in cmd[4:]))
```
Should do it. We want each element of the array to be treated as a separate argument by the shell. The shell interprets the characters contained in single quotes as single word so they're passed as a single argument.